### ISSUE TITLE
Comparison method violates its general contract

### ISSUE THREAD

**Issue by tomtom5152** [akeranen/the-one#39]
I'm getting a weird error occurring when running the simulation with any of the Prophet or MaxProp routers. It appears to happen at random points in the simulation and I cannot spot any discernable pattern.

Full stack trace using ProphetV2Router:

```java
Exception in thread "main" java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeLo(TimSort.java:777)
	at java.util.TimSort.mergeAt(TimSort.java:514)
	at java.util.TimSort.mergeCollapse(TimSort.java:441)
	at java.util.TimSort.sort(TimSort.java:245)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1454)
	at java.util.Collections.sort(Collections.java:175)
	at routing.ProphetV2Router.tryOtherMessages(ProphetV2Router.java:300)
	at routing.ProphetV2Router.update(ProphetV2Router.java:259)
	at core.DTNHost.update(DTNHost.java:346)
	at core.World.updateHosts(World.java:186)
	at core.World.update(World.java:167)
	at gui.DTNSimGUI.runSim(DTNSimGUI.java:113)
	at ui.DTNSimUI.start(DTNSimUI.java:77)
	at core.DTNSim.main(DTNSim.java:92)
```

**Comment by julianofischer** [comment]
http://delay-tolerant-networks.blogspot.com.br/2015/07/illegalargumentexception-in-simulations.html

**Comment by kenog** [comment]
I also provided a fix for this bug in #31 a month ago. If it gets merged, the workaround from that blog post will not be necessary anymore.

**Comment by tk721** [comment]
Fixed by Pull request (#31).
